src/provider.rs

```diff
@@ -172,6 +197,199 @@ impl TableProvider for SeafowlTable {
+                    Self::parse_bytes_value(&column.max_value, data_type)?;
+                null_counts.get_mut(column.name.as_ref()).unwrap()[ind] =
+                    column.null_count;
+            }
+        }
+
+        Ok(Self {
+            partition_count,
+            schema,
+            partitions: Arc::new(partitions),
+            min_values,
+            max_values,
+            null_counts,
+        })
+    }
+
+    /// Try to deserialize min/max statistics stored as raw bytes
+    ///
+    /// ```
+    /// use std::sync::Arc;
+    /// use arrow::datatypes::DataType;
+    /// use datafusion::scalar::ScalarValue;
+    /// use seafowl::provider::SeafowlPruningStatistics;
+    ///
+    /// fn parse(value: &Arc<Option<Vec<u8>>>, dt: &DataType) -> ScalarValue {
+    ///     SeafowlPruningStatistics::parse_bytes_value(&value, &dt).unwrap()
+    /// }
+    ///
+    /// // Parse missing value into a corresponding None
+    /// let val = Arc::new(None);
+    /// assert_eq!(parse(&val, &DataType::Int16), ScalarValue::Int16(None));
+    /// assert_eq!(parse(&val, &DataType::Boolean), ScalarValue::Boolean(None));
+    ///
+    /// // Parse some actual value
+    /// let val = Arc::from(Some(42.to_string().as_bytes().to_vec()));
+    /// assert_eq!(parse(&val, &DataType::Int32), ScalarValue::Int32(Some(42)));
+    /// assert_eq!(parse(&val, &DataType::Float32), ScalarValue::Float32(Some(42.0)));
+    /// assert_eq!(parse(&val, &DataType::Utf8), ScalarValue::Utf8(Some("42".to_string())));
+    /// ```
+    pub fn parse_bytes_value(
+        bytes_value: &Arc<Option<Vec<u8>>>,
+        data_type: &DataType,
+    ) -> Result<ScalarValue> {
+        match bytes_value.as_ref() {
+            Some(bytes) => match String::from_utf8(bytes.clone()) {
+                Ok(string_val) => ScalarValue::try_from_string(string_val, data_type),
+                Err(error) => Err(DataFusionError::Internal(format!(
+                    "Failed to parse min/max value: {}",
+                    error
+                ))),
+            },
+            None => {
+                // Try to cast the None value to the corresponding ScalarValue matching `data_type`,
+                // e.g. `ScalarValue::Int16(None)`, `ScalarValue::Boolean(None)`, etc.
+                let value = ScalarValue::Null;
+                let cast_options = CastOptions { safe: false };
+                let cast_arr =
+                    cast_with_options(&value.to_array(), data_type, &cast_options)?;
+                Ok(ScalarValue::try_from_array(&cast_arr, 0)?)
+            }
+        }
+    }
+
+    // Prune away partitions that are refuted by the provided filter expressions
+    async fn prune(&self, filters: &[Expr]) -> Vec<SeafowlPartition> {
+        let mut partitions = self.partitions.deref().clone();
+
+        if filters.is_empty() {
+            // No qualifiers means we need all partitions
+            return partitions;
+        }
```

Comment:
Theoretically you don't need this any more, as no filters -> partition_mask is all Trues -> will return the existing partitions. Might have a performance impact though (less so if you do iter + filter instead of clone + retain):

```suggestion
```